Sources/MongoSwift/SDAM.swift

```diff
@@ -226,9 +231,9 @@ public struct TopologyDescription {
         let timeoutValues = self.servers.map { $0.logicalSessionTimeoutMinutes }
         if timeoutValues.contains (where: { $0 == nil }) {
             return nil
-        } else {
-            return timeoutValues.map { $0! }.min()
         }
+        // swiftlint:disable:next force_unwrapping - we filtered out nil values already.
+        return timeoutValues.map { $0! }.min()
```

Comment:
Once we drop swift 4.0 we can use `compactMap` to make this a little cleaner and omit the force unwrap.
Alternatively, we can use `reduce` + length comparison above instead of `contains(where:)` which would also allow us to avoid the force unwrap, though I'm not sure there's a ton of readability / style gains there...

Reply:
good points. for now I'll add an `#if swift(..)` check here so we can `compactMap` in 4.1+ and easily switch to it once 4.0 is gone.